### PR TITLE
Import vs props

### PR DESCRIPTION
**Pull request by mradenovic**
@amr08 I tweaked your branch to make it easier to merge it with code in testing branch `add-enzyme-testing`.

* `version` and `variadic` are constants from our point of view. There is no need to pass those as props if they can be imported.
* No need to lift state up unless really necessary.
* Dropdown belongs to CardList since it is dealing with Carousel
* Commits should be atomic and revertable if possible.

### CODE REVIEW

**Comment by amr08** [comment]
@mradenovic Thanks for your input, I do have to kindly disagree with you on some of these points.   Specifically around foundational concepts of parent and child components.  

1) App is the main parent component acting as a hub.  In this case, variadic and version go to several components and are worked with, I would say it is more maintainable to be passed down directly from the hub component rather than importing the same data in several times in different files.  

2) I didn't see state being lifted unnecessarily - The parent needed access to data to pass to another child so I brought it up and passed it back down to the other child.  With that being said, Dropdown is debatable as it could live somewhere else (maybe even navbar) - I didn't think of it as tied to the carousel when I was working on it, but I don't have any strong opinions on that one to be honest.

In summary, although it might be easier (when developing) to pass state and props between the closest components or whichever one's seem convenient, in large applications it makes it nearly impossible to maintain because you don't know where anything is coming from - You don't know which component initiated state or who passed props, it's bonkers.   Anyway, these are pitfalls I've come across in my time so just trying to give it some focus/thought from the get go on the smaller projects I work on.  

Here's another good resource on parent/child/sibling that I just bookmarked: https://medium.com/@ruthmpardee/passing-data-between-react-components-103ad82ebd17

Anyway, I'm sure this wont change your mind, but I had to point it out!   🙂

**Comment by amr08** [comment]
@mradenovic To your other points, thanks for feedback on commits. I definitely get trigger happy - I think i'm done with the code, and then change something small and recommit!  I'll try to be more mindful of that moving forward.  I honestly know very little about testing, so I'll be paying attention to what you guys are doing with that.

**Comment by jhowardjr** [comment]
@amr08 @mradenovic a little peaceful discourse sounds like progress to me!

**Comment by mradenovic** [comment]
> I didn't see state being lifted unnecessarily

You are right about this. The state was related to Dropdown, and I had thoughts of having it in NavBar too. :blush:

> variadic and version go to several components and are worked with, I would say it is more maintainable to be passed down directly from the hub component rather than importing the same data several times in different files.

`version` and `variadic` are completely independent of anything else. it would be the same as passing Card, Modal and Button from App to FuncCard. If there is no big performance cost, I would go for import where I need it.

I've just started leaning React, all I know about testing is that it's very important, and this is my first group project. Any disagreement is expected (if not requested) and very much appreciated! Please keep disagreeing (now officially requested :smile: )! Thanks a lot, @amr08!

**Comment by amr08** [comment]
@mradenovic @jhowardjr haha yea exactly, agreed! No echo chambers here in the variadic group! 😛   

Diving back in, I viewed `variadic` and `version` as "data" rather than UI components.  In redux you pass down store data in a similar way even though it's totally possible to use "connect" in every component (which would make it easy to develop with, but bad practice).   Either way, for the purposes of this, I can definitely see how we could just import them.  I haven't ran this branch, but if someone has and it works feel free to merge!

**Comment by amr08** [comment]
Also @mradenovic  coming through with some epic commits! 
screen shot 2018-02-22 at 3 56 20 pm
😄
